portal-web/docroot/html/taglib/aui/button/end.jsp

```diff
@@ -22,20 +22,46 @@
 
 <c:choose>
 	<c:when test="<%= Validator.isNotNull(escapedHREF) %>">
-		<a
-			class="<%= AUIUtil.buildCss(AUIUtil.BUTTON_PREFIX, disabled, false, false, cssClass) %>"
-			href="<%= escapedHREF %>"
-			id="<%= id %>"
-
-			<c:if test="<%= Validator.isNotNull(onClick) %>">
-				onClick="<%= onClick %>"
-			</c:if>
-
-			role="button"
-
-			<%= AUIUtil.buildData(data) %>
-			<%= InlineUtil.buildDynamicAttributes(dynamicAttributes) %>
-		>
+		<c:choose>
+			<c:when test='<%= Validator.isNull(onClick) && type.equals("cancel") %>'>
+				<button
+					class="<%= AUIUtil.buildCss(AUIUtil.BUTTON_PREFIX, disabled, false, false, cssClass) %>"
+
+					<c:if test="<%= disabled %>">
+						disabled
+					</c:if>
+
+					id="<%= id %>"
+
+					<c:if test="<%= Validator.isNotNull(name) %>">
+						name="<%= namespace %><%= name %>"
+					</c:if>
+
+					onClick="Liferay.Util.navigate('<%= escapedHREF %>')"
+
+					type="button"
+
+					<%= AUIUtil.buildData(data) %>
+					<%= InlineUtil.buildDynamicAttributes(dynamicAttributes) %>
+				>
+			</c:when>
+			<c:otherwise>
+				<a
+					class="<%= AUIUtil.buildCss(AUIUtil.BUTTON_PREFIX, disabled, false, false, cssClass) %>"
+					href="<%= escapedHREF %>"
+					id="<%= id %>"
+
+					<c:if test="<%= Validator.isNotNull(onClick) %>">
+						onClick="<%= onClick %>"
+					</c:if>
+
+					role="button"
+
+					<%= AUIUtil.buildData(data) %>
+					<%= InlineUtil.buildDynamicAttributes(dynamicAttributes) %>
+				>
+			</c:otherwise>
+		</c:choose>
```

Comment:
This seems unnecessary, as we already have [the markup logic for buttons](https://github.com/liferay/liferay-portal/blob/master/portal-web/docroot/html/taglib/aui/button/end.jsp#L40-L63). Can we simplify this by just expanding `Validator.isNotNull(escapedHREF)` check and `onClick` logic?

Reply:
Nice catch! I changed it 😄